File: .codegen/templates/service.ts
import { DomainNameRepository } from '../repository/DomainNameRepository';
import { DomainName } from '../entity/DomainName';
import { Service, Inject } from 'typedi';

@Service()
export class DomainNameService {

  @Inject()
  domainNameRepository: DomainNameRepository;
  
  async getAllDomainName(): Promise<DomainName[]> {
    return this.domainNameRepository.getAllDomainName();
  }

  async getDomainNameById(id: number): Promise<DomainName|null> {
    return this.domainNameRepository.getDomainNameById(id);
  }

  async insertDomainName(domainName: DomainName): Promise<DomainName> {
    return this.domainNameRepository.insertDomainName(domainName);
  }

  async updateDomainName(domainName: DomainName): Promise<DomainName> {
    return this.domainNameRepository.updateDomainName(domainName);
  }

  async deleteDomainName(id: number) {
    this.domainNameRepository.deleteDomainName(id);
  }

}

File: .codegen/templates/service.java
package com.example.demo.service;

import com.example.demo.domain.model.DomainName;
import com.example.demo.repository.DomainNameRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class DomainNameService {

    @Autowired
    private DomainNameRepository domainNameRepository;

    public Optional<DomainName> createDomainName(DomainName domainName) {
        return Optional.of(this.domainNameRepository.save(domainName));
    }

    public Optional<DomainName> updateDomainName(DomainName domainName) {
        return Optional.of(this.domainNameRepository.save(domainName));
    }

    public List<DomainName> getAllDomainNames() {
        return this.domainNameRepository.findAll();
    }

    public Optional<DomainName> getDomainNameById(Long id) {
        return this.domainNameRepository.findById(id);
    }

    public void deleteDomainName(Long id) {
        this.domainNameRepository.deleteById(id);
    }

}
